python/redmonster/physics/zpicker2.py

```diff
@@ -107,7 +108,7 @@ def classify_obj(self, zfindobjs, zfitobjs, flags, rchi2threshold=0.005):
             npixsteptuple = ()
             fstuple = ()
             # Catch spectra that are all 0's and return null result
-            if len(n.where(self.flux[ifiber] != 0)[0]) == 0 or len(n.where(self.ivar[ifiber] != 0)[0]) == 0:
+            if len(n.where(self.flux[0] != 0)[0]) == 0:
```

Comment:
This needs to be reverted to the way it is in my copy. With this edit, it will not properly handle spectra that are all zeros if they aren't the first spectra in the array. Also, why was the check for all(ivar) == 0 removed?